docs/spec/spec.md

```diff
@@ -0,0 +1,416 @@
+
+The conforming implementation of the specification is released and included in the distribution. Any deviation from the specification is considered a bug.
+
+## Contents
+
+1. [Overview](#1-overview)
+2. [Components](#2-components)
+   - 2.1 [RFC Client](#21-rfc-client)
+     - 2.1.1 [Configurations](#211-configurations)
+       - 2.1.1.1 [DestinationConfig](#2111-destinationconfig)
+       - 2.1.1.2 [AdvancedConfig](#2112-advancedconfig)
+     - 2.1.2 [Available Operations](#212-available-operations)
+       - 2.1.2.1 [Execute a Function Module via RFC](#2121-execute-a-function-module-via-rfc)
+       - 2.1.2.2 [Send IDocs to an SAP System](#2122-send-idocs-to-an-sap-system)
+     - 2.1.3 [Data Types](#213-data-type-mapping-between-ballerina-and-sap-jco)
+   - 2.2 [Listener](#22-listener)
+     - 2.2.1 [Configurations](#221-configurations)
+       - 2.2.1.1 [ServerConfig](#2211-serverconfig)
+       - 2.2.1.2 [AdvancedConfig](#2212-advancedconfig)
+     - 2.2.2 [Available Operations](#222-available-operations)
+       - 2.2.2.1 [Receive IDocs](#2221-receive-idocs)
+       - 2.2.2.2 [Error Handling in Listener](#2222-error-handling)
+
+## 1. Overview
+
+The Ballerina SAP JCo Connector provides seamless integration with SAP systems through Java Connector (JCo) capabilities. This connector allows Ballerina applications to interact with SAP systems, enabling operations like Remote Function Call (RFC) execution, IDoc processing, and more. By leveraging this connector, developers can easily integrate SAP functionalities into their Ballerina applications, making it an essential tool for enterprises working with SAP.
+
+The `ballerinax/sap.jco` package exposes the SAP JCo library as Ballerina functions, enabling easy access to SAP's enterprise resource planning (ERP) software and other SAP solutions, such as human capital management (HCM), customer relationship management (CRM), and supply chain management (SCM).
+
+### Architecture Overview
+
+The SAP JCo Connector architecture comprises key components facilitating communication between Ballerina applications and SAP systems. These components include the SAP function modules, the SAP system itself, the Ballerina code that handles the RFC calls, and the IDoc servers responsible for sending and receiving IDocs. This architecture enables robust integration, supporting various enterprise operations.
+
+## 2. Components
+
+This section outlines the core components of the Ballerina SAP JCo Connector and their configurations, focusing on how they interact with the SAP system.
+
+### 2.1 RFC Client
+
+The RFC Client component allows the Ballerina application to communicate with an SAP system by invoking remote function modules and sending IDocs.
+
+#### 2.1.1 Configurations
+
+The RFC Client supports two types of configurations, `DestinationConfig` and `AdvancedConfig`, to establish a connection with the SAP system.
+
+##### 2.1.1.1 DestinationConfig
```

Comment:
Wouldn't it be better to have the title as `Destination configurations` (instead of having just the record typename) for better readability?
```suggestion
##### 2.1.1.1 DestinationConfig
```